packages/graphql-codegen-cli/src/utils/watcher.ts

```diff
@@ -99,22 +75,20 @@ export const createWatcher = (
       async (_, events) => {
         // it doesn't matter what has changed, need to run whole process anyway
         await Promise.all(
+          // NOTE: @parcel/watcher always provides path as an absolute path
           events.map(async ({ type: eventName, path }) => {
-            /**
-             * @parcel/watcher has no way to run watcher on specific files (https://github.com/parcel-bundler/watcher/issues/42)
-             * But we can use micromatch to filter out events that we don't care about
-             */
-            if (!mm.contains(path, files)) return;
+            if (!shouldRebuild({ path })) {
+              return;
+            }
```

Comment:
This is the crux of the matter: we call the previously constructed `shouldRebuild()` function on change events.